### PR TITLE
API: Adding `featured` filter option to posts.browse

### LINKED ISSUE #5152: API: Add `featured` filter option to posts.browse
Currently the `api.posts.browse` endpoint takes the [following options](https://github.com/TryGhost/Ghost/blob/master/core/server/api/posts.js#L48):

> page, limit, status, staticPages, tag

There is an open issue to swap [`page` for `offset`](https://github.com/TryGhost/Ghost/issues/5093)

And another issue to swap [`staticPages` for `page`](https://github.com/TryGhost/Ghost/issues/5151) as well as add generally better support for querying pages.

It would also be good to be able to filter posts by whether or not they are featured: 

`api.posts.browse({featured: true})` or `api.posts.browse({featured: false})` 

Making this work involves adding this as an allowed option in the model.

### PR DESCRIPTION
**Pull request by edsadr**
Closes #5152
- Adds `featured` filter option to posts.browse method modifying the model to take it too

### CODE REVIEW

**Comment by edsadr** [comment]
@ErisDS  Any update on this one? waiting this one to be merged to move on to the next change

File: core/server/api/posts.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ var Promise         = require('bluebird'),
     utils           = require('./utils'),
 
     docName         = 'posts',
-    allowedIncludes = ['created_by', 'updated_by', 'published_by', 'author', 'tags', 'fields', 'next', 'previous'],
+    allowedIncludes = ['created_by', 'updated_by', 'published_by', 'author', 'tags', 'fields', 'next', 'previous', 'featured'],
```
Comment (ErisDS) [inline core/server/api/posts.js]: As `featured` doesn't represent a related model, it doesn't need to be listed here.

**Comment by ErisDS** [comment]
Sorry for the delay, left a comment on one line but this is otherwise looking good to go :+1:

File: core/server/api/posts.js (hunk 1 above)
Comment (edsadr) [inline core/server/api/posts.js]: Ok is removed, sorry I misunderstood that part

**Comment by ErisDS** [comment]
Sorry, just spotted the commit message - can you please update it to match our [guidelines](https://github.com/TryGhost/Ghost/wiki/Git-workflow#commit-messages)? It needs to please have a short description as the first line - that's what goes into the changelog.

**Pull request merged**